netty/src/main/java/io/grpc/netty/NettyClientHandler.java

```diff
@@ -467,6 +469,25 @@ private void gracefulClose(ChannelHandlerContext ctx, GracefulCloseCommand msg,
     close(ctx, promise);
   }
 
+  private void forcefulClose(final ChannelHandlerContext ctx, final ForcefulCloseCommand msg,
+      ChannelPromise promise) throws Exception {
+    lifecycleManager.notifyShutdown(
+        Status.UNAVAILABLE.withDescription("Channel requested transport to shut down"));
```

Comment:
Good eye. This is a bit of slight of hand...

I was thinking it would probably be useful to know that a pre-existing RPC was cancelled because of shutdown now, instead of the RPC simply starting after shutdown. To do that "correctly" I would probably want this status to come from NettyClientTransport like is done for `shutdown()`, but figuring out a name and doing the plumbing was annoying and it wasn't clear whether this was beneficial behavior.

And now that I'm thinking about it more, I'm even questioning whether we should use `Status.CANCELLED.withDescription("Server shutdownNow invoked");` instead of `UNAVAILABLE` when we kill the streams...